coordinator/schedulepod/main.go

```diff
@@ -0,0 +1,694 @@
+		return err
+	}
+	featureID := metadata.ResourceID{Name: featureName, Variant: "", Type: metadata.FEATURE_VARIANT}
+	featureCreated, err := client.GetFeatureVariant(context.Background(), metadata.NameVariant{Name: featureName, Variant: ""})
+	if err != nil {
+		return fmt.Errorf("could not get feature: %v", err)
+	}
+	if featureCreated.Status() != metadata.CREATED {
+		return fmt.Errorf("Feature not set to created with no coordinator running")
+	}
+	kubeJobSpawner := coordinator.KubernetesJobSpawner{}
+	coord, err := coordinator.NewCoordinator(client, logger, cli, &kubeJobSpawner)
+	if err != nil {
+		return fmt.Errorf("Failed to set up coordinator")
+	}
+	go func() {
+		if err := coord.WatchForNewJobs(); err != nil {
+			logger.Errorf("Error watching for new jobs: %v", err)
+		}
+	}()
+	go func() {
+		if err := coord.WatchForUpdateEvents(); err != nil {
+			logger.Errorf("Error watching for new update events: %v", err)
+		}
+	}()
+	time.Sleep(70 * time.Second)
+	jobClient, err := runner.NewKubernetesJobClient(runner.GetCronJobName(featureID), runner.Namespace)
+	if err != nil {
+		return err
+	}
+	cronJob, err := jobClient.GetCronJob()
+	if err != nil {
+		return err
+	}
+	lastExecutionTime := cronJob.Status.LastSuccessfulTime
+	if lastExecutionTime.IsZero() {
+		return fmt.Errorf("job did not execute in time")
+	}
+	// featureUpdated, err := client.GetFeatureVariant(context.Background(), metadata.NameVariant{Name: featureID.Name, Variant: ""})
+	// if err != nil {
+	// 	return err
+	// }
+	// featureUpdateStatus := featureUpdated.UpdateStatus()
+	// featureLastUpdated := featureUpdateStatus.LastUpdated
+	// if featureLastUpdated.AsTime().IsZero() {
+	// 	return fmt.Errorf("Scheduler did not update feature")
+	// }
+	return nil
+}
+
+func testScheduleTransformation() error {
+	if err := runner.RegisterFactory(string(runner.CREATE_TRANSFORMATION), runner.CreateTransformationRunnerFactory); err != nil {
+		return fmt.Errorf("Failed to register training set runner factory: %v", err)
+	}
+	defer runner.UnregisterFactory(string(runner.CREATE_TRANSFORMATION))
+	logger := zap.NewExample().Sugar()
+	client, err := metadata.NewClient(metadata_addr, logger)
+	if err != nil {
+		return fmt.Errorf("Failed to connect: %v", err)
+	}
+	defer client.Close()
+	etcdConnect := fmt.Sprintf("%s:%s", etcdHost, etcdPort)
+	cli, err := clientv3.New(clientv3.Config{Endpoints: []string{etcdConnect}})
+	if err != nil {
+		return err
+	}
+	defer cli.Close()
+	tableName := createSafeUUID()
+	serialPGConfig := postgresConfig.Serialize()
+	if err != nil {
+		return fmt.Errorf("could not get provider as offline store: %v", err)
+	}
+	if err := CreateOriginalPostgresTable(tableName); err != nil {
+		return fmt.Errorf("Could not create non-featureform source table: %v", err)
+	}
+	sourceName := strings.Replace(createSafeUUID(), "-", "", -1)
```

Comment:
Why do we need to do this?

Reply:
this bit `strings.Replace(createSafeUUID(), "-", "", -1)`

Reply:
it's because of the issue with dashes in kubernetes jobs